File: tests/adapters/rust/goldens/complex/max_optimization.rs
// … comment omitted
// … 3 imports omitted

// … comment omitted
// … 3 imports omitted (2 lines)

// … comment omitted
// … 2 imports omitted

/// Module level long documentation that might be truncated under tight budgets.
// … 2 consts omitted (4 lines)

// … macro omitted (13 lines)

// … struct omitted (6 lines)

/// PublicService provides public API operations.
pub struct PublicService {
    // … field omitted
}

impl PublicService {
    /// Creates a new service instance.
    pub fn new() -> Self {
        // … method body omitted (3 lines)
    }

    /// GetUser is a public API method that gets a user by ID.
    pub fn get_user(&self, id: i32) -> Option<&User> {
        // … method body omitted (2 lines)
    }

    /// normalize is a private helper — should not be visible with public_api_only.
    // … method omitted (5 lines)

    /// Process is a long method body to allow function body stripping.
    pub fn process(&mut self, list: Vec<User>) -> Result<Vec<User>> {
        // … method body omitted (9 lines)
    }
}

/// internalOnly is a private struct — should be filtered out in public_api_only.
// … struct omitted (3 lines)

// … impl omitted (5 lines)

/// PublicFunction is an exported function.
pub fn public_function(name: String) -> String {
    // … function body omitted (2 lines)
}

// … 2 functions omitted (12 lines)

File: tests/adapters/python/goldens/complex/max_optimization.py
"""Comprehensive sample for Budget System tests (Python)."""

# … 13 imports omitted (12 lines)


MODULE_DOC = """
This module demonstrates a variety of language features to exercise the
BudgetController escalation sequence. The text here is quite verbose and
contains enough co…""" # literal string (−41 tokens)

BIG_LIST = [f"item_{i:04d}" for i in range(200)]

BIG_DICT = {
    "users": [{"id": i, "name": f"User {i}", "active": i % 2 == 0} for i in range(50)],
    # … (1 more, −60 tokens)
}


def public_function(data: str) -> str:
    """Public API function."""
    # … function body omitted (2 lines)


# … function omitted (6 lines)


class PublicClass:
    """Public class exposed to users."""

    def __init__(self, name: str):
        # … method body omitted (2 lines)

    def public_method(self, x: int, y: int) -> int:
        """Add two numbers and return the result."""
        return x + y

    # … method omitted (3 lines)

    @property
    def public_property(self) -> str:
        """Public property."""
        return self.name

    # … method omitted (4 lines)


# … class omitted (4 lines)


def huge_processing_pipeline(values: List[int]) -> Tuple[int, int, int]:
    """A long body that can be stripped when budgets are tight."""
    # … function body omitted (9 lines)


def another_long_function():
    """Another long function body to enable stripping heuristics."""
    # … function body omitted (4 lines)


if __name__ == "__main__":
    inst = PublicClass("demo")
    print(public_function("hello"))
    print(inst.public_method(2, 3))
